pkg/cloud/generic/actuators/machine/actuator.go

```diff
@@ -52,32 +52,72 @@ func NewActuator(params ActuatorParams) (*Actuator, error) {
 
 // Create creates a machine and is invoked by the Machine Controller
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	// Create is only called if a Machine resource exists. It is a
+	// configuration error if a Cluster does not exist in the same
+	// namespace as the Machine. By returning an error here we ensure
+	// the resource will be requeued for processing later.
+	if cluster == nil {
+		return fmt.Errorf("machine %s does not have cluster in namespace %s", machine.Name, machine.Namespace)
+	}
+
 	log.Printf("Creating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
+	providerID := fmt.Sprintf("%s-%s", cluster.Name, machine.Name)
+	machine.Spec.ProviderID = &providerID
+	machine, err := a.machinesGetter.Machines(machine.Namespace).Update(machine)
```

Comment:
It might be good to do a `Patch` here instead of `Update` to avoid issues related resourceVersion being revved elsewhere.